### PR TITLE
reference/sql: add document for expression index

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What is changed, added or deleted? 
Add document for expression index.

### What is the related PR or file link(s)? 

- Reference link(s):
- This PR is to align with:
- [x] N/A (not applicable)

**Important Notice:**

**If your changes apply to multiple TiDB documentation versions**, to trigger the bot to cherry-pick this PR to other release branches, make sure you add labels such as:

- **needs-cherry-pick-3.0**
- **needs-cherry-pick-3.1**
- **needs-cherry-pick-2.1**

### CODE REVIEW

File: reference/sql/statements/create-index.md [hunk 1]
```diff
@@ -128,6 +128,18 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
...
+
+TiDB 支持创建表达式索引，即将索引建立在一个表达式上。示例如下：
+
+{{< copyable "sql" >}}
+
+```sql
+CREATE INDEX c1_expr ON t1 ((c1+1));
+```
+
+表达式索引的语法和限制与 MySQL 相同，是通过将索引建立在隐藏的虚拟列上来实现的。因此所支持的表达式继承了虚拟列的所有限制。目前，建立了索引的表达式只有在 `FIELD` 子句、`WHERE` 子句和 `ORDER BY` 子句中时，优化器才能使用表达式索引。后续将支持 `GROUP BY` 子句。
```
Comment (Deardrops) [inline reference/sql/statements/create-index.md]: > 所支持的表达式继承了虚拟列的所有限制

这里 `虚拟列的所有限制` 可以加个链接，到生成列页面。

Comment (Deardrops) [reply reference/sql/statements/create-index.md]: 虚拟列其实并没有在我们文档中出现，文档中的叫法是 `Generated virtual column`

File: reference/sql/statements/create-index.md [hunk 2]
```diff
@@ -128,6 +128,18 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
 Query OK, 0 rows affected (0.31 sec)
 ```
 
+## 表达式索引
+
+TiDB 支持创建表达式索引，即将索引建立在一个表达式上。示例如下：
```
Comment (Deardrops) [inline reference/sql/statements/create-index.md]:
```suggestion
TiDB 支持创建表达式索引，即以表达式作为索引项。当数据查询语句中出现已创建索引的表达式时，TiDB 可以使用已创建的索引加速查询。示例如下：
```

File: reference/sql/statements/create-index.md [hunk 3]
```diff
@@ -128,6 +128,30 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
 Query OK, 0 rows affected (0.31 sec)
 ```
 
+## 表达式索引
+
+TiDB 不仅能将索引建立在表中的一个或多个列上，还可以将索引建立在一个表达式上。当查询涉及表达式时，表达式索引能够加速这些查询。示例如下：
```
Comment (TomShawn) [inline reference/sql/statements/create-index.md]:
```suggestion
TiDB 不仅能将索引建立在表中的一个或多个列上，还可以将索引建立在一个表达式上。当查询涉及表达式时，表达式索引能够加速这些查询。
```

File: reference/sql/statements/create-index.md [hunk 4]
```diff
@@ -128,6 +128,30 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
...
+
+考虑以下查询：
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t WHERE lower(name) = "pingcap";
+```
+
+如果建立了如下的表达式索引，上述的查询就可以使用索引进行加速：
```
Comment (TomShawn) [inline reference/sql/statements/create-index.md]:
```suggestion
如果建立了如下的表达式索引，就可以使用索引加速以上查询：
```

File: reference/sql/statements/create-index.md [hunk 5]
```diff
@@ -128,6 +128,30 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
...
+
+{{< copyable "sql" >}}
+
+```sql
+CREATE INDEX idx ON t ((lower(name)));
+```
+
+维护表达式索引的代价比一般的索引更高，因为在插入或者更新每一行时都需要计算出表达式的值。然而，因为表达式的值已经存储在索引中，所以当优化器选择表达式索引时，表达式的值就不需要再计算。因此，当查询速度比插入速度和更新速度更重要时，可以考虑建立表达式索引。
```
Comment (TomShawn) [inline reference/sql/statements/create-index.md]:
```suggestion
维护表达式索引的代价比一般的索引更高，因为在插入或者更新每一行时都需要计算出表达式的值。因为表达式的值已经存储在索引中，所以当优化器选择表达式索引时，表达式的值就不需要再计算。因此，当查询速度比插入速度和更新速度更重要时，可以考虑建立表达式索引。
```

File: reference/sql/statements/create-index.md [hunk 6]
```diff
@@ -128,6 +128,30 @@ CREATE UNIQUE INDEX c1 ON t1 (c1);
...
+
+表达式索引的语法和限制与 MySQL 相同，是通过将索引建立在隐藏的虚拟生成列(generated virtual column)上来实现的。因此所支持的表达式继承了虚拟生成列的所有[限制](/reference/sql/generated-columns.md#局限性)。目前，建立了索引的表达式只有在 `FIELD` 子句、`WHERE` 子句和 `ORDER BY` 子句中时，优化器才能使用表达式索引。后续将支持 `GROUP BY` 子句。
```
Comment (TomShawn) [inline reference/sql/statements/create-index.md]:
```suggestion
表达式索引的语法和限制与 MySQL 相同，是通过将索引建立在隐藏的虚拟生成列 (generated virtual column) 上来实现的。因此所支持的表达式继承了虚拟生成列的所有[限制](/reference/sql/generated-columns.md#局限性)。目前，建立了索引的表达式只有在 `FIELD` 子句、`WHERE` 子句和 `ORDER BY` 子句中时，优化器才能使用表达式索引。后续将支持 `GROUP BY` 子句。
```

**Pull request merged**